app/services/course/assessment/submission/update_service.rb

```diff
@@ -3,6 +3,11 @@ class Course::Assessment::Submission::UpdateService < SimpleDelegator
   include Course::Assessment::Answer::UpdateAnswerConcern
 
   def update
+    is_course_koditsu_enabled = @assessment.course.component_enabled?(Course::KoditsuPlatformComponent)
+    is_assessment_koditsu_error = @assessment.is_koditsu_enabled && !@assessment.koditsu_assessment_id
+
+    raise KoditsuError if is_course_koditsu_enabled && is_assessment_koditsu_error
```

Comment:
In what workflow can there be an update to a submission which `is_koditsu_enabled`, but does not have a corresponding `koditsu_assessment_id`?  
If it's an impossible state, then can remove this, prefer not to incur an additional `@assessment.course` relation query in the submission update.

Reply:
you're right. I was placing the logic here for being secure, as I was thinking if it's possible to have `koditsu_assessment_id` to be `nil` even when submission already exists.

Turns out, this will never happen, and I have removed the logic accordingly